crates/biome_js_analyze/src/lint/nursery/no_require_imports.rs

```diff
@@ -0,0 +1,74 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_js_syntax::JsCallExpression;
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Use the newer ES6-style imports over require().
+    ///
+    /// ES6-style `import`s are modern alternative to CommonJS `require` imports. Supported by all modern browsers and Node.js versions.
+    /// Tooling can more easily statically analyze and tree-shake ES6-style `import`s compared in comparison to `require`.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// require('node:fs');
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// import fs from 'node:fs';
+    /// ```
+    /// ```js
+    /// import('node:fs')
+    /// ```
+    ///
+    pub NoRequireImports {
+        version: "next",
+        name: "noRequireImports",
+        language: "js",
+        sources: &[
+            RuleSource::EslintTypeScript("no-require-imports"),
+        ],
+        recommended: false,
+    }
+}
+
+impl Rule for NoRequireImports {
+    type Query = Ast<JsCallExpression>;
```

Comment:
You still need to use the semantic model, and make sure that `require` is a global. If it's not a global, the rule should not trigger.

For example 

```js
const require = () => {};
require()
```